app/webpacker/lib/hooks/useStoredReducer.js

```diff
@@ -0,0 +1,33 @@
+import { useReducer } from 'react';
+
+/**
+ * This functions like the useReducer hook, but it fetches the state stored in
+ * local storage, via the given key, on subsequent uses.
+ *
+ * Do NOT call this twice with the same key - updating one of such a pair
+ * will not update the other's state.
+ */
+export default function useStoredReducer(reducer, initialState, key) {
+  let storedState;
+  try {
+    storedState = JSON.parse(localStorage.getItem(key));
+  } catch {
+    storedState = null;
+  }
```

Comment:
Under which circumstances can an error occur? I would prefer a "is `localStorage.getItem(key)` present"-style check over a carte blanche catch approach

Reply:
`JSON.parse` will throw an exception if it can't parse its argument, which in theory could be anything since it's coming from local storage. The try-catch isn't for `localStorage.getItem`.